system/CodeIgniter.php

```diff
@@ -880,7 +890,9 @@ protected function display404errors(PageNotFoundException $e)
             ob_end_flush(); // @codeCoverageIgnore
         }
 
-        throw PageNotFoundException::forPageNotFound(ENVIRONMENT !== 'production' || is_cli() ? $e->getMessage() : '');
+        throw PageNotFoundException::forPageNotFound(
+            ENVIRONMENT !== 'production' || ! ($this->context === 'web') ? $e->getMessage() : ''
```

Comment:
It seems needs parentheses to avoid confusion when read:

```suggestion
            ENVIRONMENT !== 'production' || ($this->context !== 'web' ? $e->getMessage() : '')
```

and possibly cause different result of expected output, ref https://3v4l.org/jNQEV vs https://3v4l.org/EqcGv

Reply:
It should be:
```php
            (ENVIRONMENT !== 'production' || ! ($this->context === 'web')) ? $e->getMessage() : ''
```
```php
            (ENVIRONMENT !== 'production' || $this->context !== 'web') ? $e->getMessage() : ''
```

Reply:
This one is painful to read on mobile so I'll let you two figure it out 😅

Reply:
Added parentheses,